packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -905,7 +906,15 @@ function collectParams(
       if (value.type.kind === "Model") {
         collectParams(value.type.properties, params);
       } else {
-        params.push(value);
+        let sourceProp = value;
+        while (sourceProp.sourceProperty) {
+          sourceProp = sourceProp.sourceProperty;
+        }
+        if (sourceProp.model && !isAzureCoreModel(sourceProp.model)) {
```

Comment:
Can we add a log when we skip core models? It shouldn't happen often, but when it does, it might confuse the user. So, having a log might be useful.